### PR TITLE
Light saber

### PR DESCRIPTION
**Pull request by Arielelnoob115Z**

### CODE REVIEW

File: Assets/Scripts/Combat/LightSaber.cs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    [SerializeField] private PlayableDirector _lightSaberOut;
+    [SerializeField] private PlayableDirector _lightSaberIn;
+    [Header("GameObject Related")]
+    [SerializeField] private Collider _trackEnemiesZone;
+
+    #endregion
+
+    #region Private Variables
+
+    int _enemyIndex = 0;
```
Comment (RickyMorel) [inline Assets/Scripts/Combat/LightSaber.cs]: private

File: Assets/Scripts/Interactable/Weapon.cs [hunk 2]
```diff
@@ -28,6 +29,12 @@ public class Weapon : Upgradable
 
     #endregion
 
+    #region Getters and Setters
+
+    public bool ShouldRotate { get { return _shouldRotate; } set { _shouldRotate = value; } }
```
Comment (RickyMorel) [inline Assets/Scripts/Interactable/Weapon.cs:34]: ?

File: Assets/Scripts/AI/AIHealth.cs [hunk 3]
```diff
@@ -61,6 +68,16 @@ private void HandleDead()
         Invoke(nameof(DisableSelf), 10f);
     }
 
+    private void CutMeshIfLightSaber()
+    {
+        if (!(_damageData.DamageTypes[0] == DamageTypes.Base && _damageData.DamageTypes[1] == DamageTypes.Laser)) { return; }
+        if (!(_damageData.DamageTypes[1] == DamageTypes.Laser && _damageData.DamageTypes[0] == DamageTypes.Base)) { return; }
```
Comment (RickyMorel) [inline Assets/Scripts/AI/AIHealth.cs]: ?

Comment (Arielelnoob115Z) [reply Assets/Scripts/AI/AIHealth.cs]: Ricky likes sissy femboys

**Review by RickyMorel — APPROVED** [review]
oIkO iTE

**Pull request merged**